### PR TITLE
Tweak Http2Error and add tests

### PR DESCRIPTION
**Pull request by ejona86**

### CODE REVIEW

File: core/src/main/java/io/grpc/transport/HttpUtil.java [hunk 1]
```diff
@@ -189,15 +189,22 @@ public static Http2Error forCode(long code) {
     }
 
     /**
-     * Looks up the {@link Status} from the given HTTP/2 error code.
+     * Looks up the {@link Status} from the given HTTP/2 error code. This is preferred over {@code
+     * forCode(code).status()}, to more easily conform to HTTP/2:
+     *
+     * <blockquote><code>Unknown or unsupported error codes MUST NOT trigger any special behavior.
```
Comment (buchgr) [inline core/src/main/java/io/grpc/transport/HttpUtil.java]: do we need blockquote and code? if yes, close blockquote?

File: core/src/test/java/io/grpc/transport/HttpUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        Http2Error.forCode(Http2Error.HTTP_1_1_REQUIRED.code()));
+    for (Http2Error error : Http2Error.values()) {
+      assertSame(error, Http2Error.forCode(error.code()));
+    }
+  }
+
+  @Test
+  public void http2ErrorNoCode() {
+    assertNull(Http2Error.forCode(-1));
+    assertNull(Http2Error.forCode(0xD + 1));
```
Comment (buchgr) [inline core/src/test/java/io/grpc/transport/HttpUtilTest.java]: Maybe add a comment explaining `-1` and `0xD + 1`?

File: core/src/test/java/io/grpc/transport/HttpUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  @Test
+  public void http2ErrorStatus() {
+    // Nothing special about this particular error, except that it is slightly distinctive.
+    assertSame(Status.Code.CANCELLED, Http2Error.CANCEL.status().getCode());
+  }
+
+  @Test
+  public void http2ErrorNoCodeStatus() {
+    assertSame(Status.Code.INTERNAL, Http2Error.statusForCode(-1).getCode());
+    assertSame(Status.Code.INTERNAL, Http2Error.statusForCode(0xD + 1).getCode());
```
Comment (buchgr) [inline core/src/test/java/io/grpc/transport/HttpUtilTest.java]: same as above.

File: core/src/test/java/io/grpc/transport/HttpUtilTest.java (hunk 2 above)
Comment (ejona86) [inline core/src/test/java/io/grpc/transport/HttpUtilTest.java]: Like "// Edge cases" ? Something more?

Comment (buchgr) [inline core/src/test/java/io/grpc/transport/HttpUtilTest.java]: `// Edge Cases. Valid error codes range from 0 to 0xD` maybe?

File: core/src/main/java/io/grpc/transport/HttpUtil.java (hunk 1 above)
Comment (ejona86) [inline core/src/main/java/io/grpc/transport/HttpUtil.java]: I guess we don't need code, since it is just text (<code> does make it look like an RFC though :-) ).

**Comment by ejona86** [comment]
@buchgr, PTAL

**Comment by buchgr** [comment]
@ejona86 LGTM
